actions/global_actions.jsx

```diff
@@ -120,11 +122,17 @@ export function showShortcutsModal() {
 }
 
 export function showGetPublicLinkModal(fileId) {
-    AppDispatcher.handleViewAction({
-        type: ActionTypes.TOGGLE_GET_PUBLIC_LINK_MODAL,
-        value: true,
-        fileId,
-    });
+    return async (dispatch) => {
+        await dispatch(getFilePublicLink(fileId));
```

Comment:
This is a slight product change (possible improvement). Before the modal could load without the link present. Now the link is populated before the modal opens. Downside is there is a longer delay for the modal popping up. The upside is that there won't be a broken link in the modal.

Reply:
Can we pass in the `fileId` to the modal and make modal responsible for `getFilePublicLink `?. This way we can retain the UX the way it was. Having modal wait on an API will have not so great consequences with network latency particularly without having a feedback that we will be opening a modal.

Reply:
If the modal opens, and the link isn't present I don't think thats great UX either. If there is high network latency both options have issues. IMO the modal should be a dumb object and just present the information. Having the modal handle the request to `getFilePublicLink(fileId)` makes it much more complicated. Happy to change the UX back.